src/dragger.js

```diff
@@ -0,0 +1,126 @@
+   * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer
+   *     has moved from the position at the start of the drag, in pixel units.
+   * @param {boolean} healStack Whether or not to heal the stack after
+   *     disconnecting.
+   */
+  startDrag(currentDragDeltaXY, healStack) {
+    if (!inMultipleSelectionMode) {
+      const blockDraggerList = [];
+      if (!blockSelection.has(this.block_.id)) {
+        blockSelection.forEach((id) => {
+          const element = this.workspace_.getBlockById(id);
+          if (!element.disposed) {
+            element.pathObject.updateSelected(false);
+          }
+        });
+        blockSelection.clear();
+        blockDraggerList.push(new Blockly.BlockDragger(this.block_,
+            this.workspace_));
+        this.block_.pathObject.updateSelected(true);
+      }
+      blockSelection.forEach((id) => {
+        const element = this.workspace_.getBlockById(id);
+        if (element.disposed) {
+          blockSelection.delete(id);
+          return;
+        }
+        // Only drag the parent if it is selected.
+        if (hasSelectedParent(element)) {
+          return;
+        }
+        blockDraggerList.push(new Blockly.BlockDragger(element,
+            this.workspace_));
+      });
+      blockDraggerList.forEach((blockDragger) => {
+        blockDragger.startDrag(currentDragDeltaXY, healStack);
+        this.blockDraggers_.add(blockDragger);
+      });
+      if (this.blockDraggers_.size > 1) {
+        // Disabled the highlighting around connection for multiple blocks
+        // dragging because of the bugs.
+        Blockly.RenderedConnection.prototype.highlight = function() {};
+      }
+    }
+  }
+
+  /**
+   * Moves the block to the specified location.
+   * @param {!Event} e The mouseup/touchend event.
+   * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer
+   *     has moved from the position at the start of the drag, in pixel units.
+   */
+  drag(e, currentDragDeltaXY) {
+    this.blockDraggers_.forEach(function(blockDragger_) {
+      blockDragger_.drag(e, currentDragDeltaXY);
+    });
+    e.preventDefault();
+    e.stopPropagation();
+  }
+
+  /**
+   * Finishes the block drag.
+   * @param {!Event} e The mouseup/touchend event.
+   * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer
+   *     has moved from the position at the start of the drag, in pixel units.
+   */
+  endDrag(e, currentDragDeltaXY) {
+    this.blockDraggers_.forEach((blockDragger_) => {
+      if (Blockly.Events.getGroup()) {
+        this.group_ = Blockly.Events.getGroup();
+      } else {
+        Blockly.Events.setGroup(this.group_);
+      }
+      blockDragger_.endDrag(e, currentDragDeltaXY);
+    });
+    if (this.blockDraggers_.size > 1) {
+      // Restore the highlighting around connection for multiple blocks.
+      Blockly.RenderedConnection.prototype.highlight = origHighlight;
```

Comment:
I'd like to find a way around this that doesn't involve monkey patching the prototype, but I'll have to think on it haha. @BeksOmega for later.

Reply:
If we can luckily fix the bug, then that monkey patch is no longer needed.

The bug is that, when you finished connecting multiple blocks at the same time, only one of the highlighted connection path that appear during the multiple drag can get deleted and others will still remain:
![](https://user-images.githubusercontent.com/43995067/178642540-8cab4af4-9512-45ca-8215-f512e671f796.png)
![](https://user-images.githubusercontent.com/43995067/178642574-c67e5423-bde6-4f84-928c-ef23ad62a813.png)
![](https://user-images.githubusercontent.com/43995067/178642589-ec4a8430-0128-4135-92c1-74adc97387d3.png)

So the patch here is just a work around, it disables showing the highlight in a multiple drag.
![image](https://user-images.githubusercontent.com/43995067/178643187-3458cd4e-d6c9-42a9-b954-8457c46f94c3.png)

Currently I have no idea on how to fix that, may be some upstream patches will be necessary I guess.